backend/src/test/kotlin/app/ehrenamtskarte/backend/auth/webservice/ApiTokenServiceTest.kt

```diff
@@ -0,0 +1,210 @@
+    private val mockContext = mockk<GraphQLContext>()
+    private val mockJwtPayload = mockk<JwtPayload>()
+
+    @BeforeEach
+    fun setUp() {
+        every { mockDfe.getContext<GraphQLContext>() } returns mockContext
+        every { mockContext.enforceSignedIn() } returns mockJwtPayload
+        transaction {
+            ApiTokens.deleteAll()
+        }
+    }
+
+    @Test
+    fun createApiToken_createsTokenSuccessfully() {
+        every { mockJwtPayload.adminId } returns TestAdministrators.KOBLENZ_PROJECT_ADMIN.id
+
+        transaction {
+            val numberOfTokensBefore = ApiTokens.selectAll().count()
+            ApiTokenService().createApiToken(3, mockDfe)
+            val numberOfTokensAfter = ApiTokens.selectAll().count()
+            assertEquals(numberOfTokensBefore + 1, numberOfTokensAfter)
+        }
+    }
+
+    @Test
+    fun createApiToken_throwsForbiddenExceptionWhenNotAuthorized() {
+        every { mockJwtPayload.adminId } returns TestAdministrators.NUERNBERG_PROJECT_STORE_MANAGER.id
+
+        transaction {
+            assertThrows(ForbiddenException::class.java) {
+                ApiTokenService().createApiToken(1, mockDfe)
+            }
+            assertFalse(
+                Authorizer.mayAddApiTokensInProject(
+                    AdministratorEntity.find { Administrators.id eq mockJwtPayload.adminId }.single()
+                )
+            )
+        }
+    }
+
+    @Test
+    fun createApiToken_throwsExceptionWhenAdminNotFound() {
+        every { mockJwtPayload.adminId } returns 1234
+
+        transaction {
+            assertThrows(ForbiddenException::class.java) {
+                ApiTokenService().createApiToken(1, mockDfe)
+            }
+        }
+    }
+
+    @Test
+    fun deleteApiToken_deletesTokenSuccessfully() {
+        every { mockJwtPayload.adminId } returns TestAdministrators.KOBLENZ_PROJECT_ADMIN.id
+
+        transaction {
+            val tokenId = TestData.createApiToken(creatorId = TestAdministrators.KOBLENZ_PROJECT_ADMIN.id)
+            val tokenExists = ApiTokens.select { ApiTokens.id eq tokenId }.count() > 0
+            assertTrue(tokenExists)
+
+            ApiTokenService().deleteApiToken(tokenId, mockDfe)
+
+            val tokenNoLongerExists = ApiTokens.select { ApiTokens.id eq tokenId }.singleOrNull() == null
+            assertTrue(tokenNoLongerExists)
+        }
+    }
+
+    @Test
+    fun deleteApiToken_deletesFromOtherAdminTokenSuccessfully() {
+        every { mockJwtPayload.adminId } returns TestAdministrators.KOBLENZ_PROJECT_ADMIN.id
+        TestData.createApiToken(creatorId = TestAdministrators.KOBLENZ_PROJECT_ADMIN_2.id)
+
+        transaction {
+            val tokenBefore = ApiTokens.select { ApiTokens.creatorId eq TestAdministrators.KOBLENZ_PROJECT_ADMIN_2.id }.singleOrNull()
+            assertNotNull(tokenBefore)
+
+            ApiTokenService().deleteApiToken(tokenBefore!!.get(ApiTokens.id).value, mockDfe)
```

Comment:
```suggestion
            ApiTokenService().deleteApiToken(tokenBefore!![ApiTokens.id].value, mockDfe)
```
Got a warning here to use indexing